addons/io_hubs_addon/components/operators.py

```diff
@@ -437,6 +437,63 @@ def split_and_prefix_report_messages(report_string):
     return [f"{i+1:02d}   {message}" for i, message in enumerate(report_string.split("\n\n"))]
 
 
+class CopyHubsComponent(Operator):
+    bl_idname = "wm.copy_hubs_component"
+    bl_label = "Copy component from active object"
+    bl_options = {'REGISTER', 'UNDO'}
+
+    panel_type: StringProperty(name="panel_type")
+    component_name: StringProperty(name="component_name")
+
+    def get_selected_bones(self, context):
+        selected_bones = context.selected_pose_bones if context.mode == "POSE" else context.selected_editable_bones
+        selected_armatures = [sel_ob for sel_ob in context.selected_objects if sel_ob.type == "ARMATURE"]
+        selected_hosts = []
+        for armature in selected_armatures:
+            armature_bones = armature.pose.bones if context.mode == "POSE" else armature.data.edit_bones
+            target_armature_bones = armature.data.bones if context.mode == "POSE" else armature.data.edit_bones
+            target_bones = [bone for bone in armature_bones if bone in selected_bones]
+            for target_bone in target_bones:
+                selected_hosts.append(*[bone for bone in target_armature_bones if target_bone.name == bone.name])
+        return selected_hosts
+
+    def get_selected_hosts(self, context):
+        selected_hosts = []
+        for host in context.selected_objects:
+            if host.type == "ARMATURE" and context.mode != "OBJECT":
+                selected_hosts.append(*self.get_selected_bones(context))
+            else:
+                selected_hosts.append(host)
+
+        return selected_hosts
+
+    def execute(self, context):
+        src_host = None
+        selected_hosts = self.get_selected_hosts(context)
```

Comment:
I think this can just be initialized to a blank list?

Reply:
Yeah we can do it